### PR TITLE
require 3.6.11 to automatically spin up workflow services (worker, backend)

### PR DESCRIPTION
**Pull request by avimoondra**
Baseline

```
➜  retool git:(main) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled=false --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool

➜  retool git:(main) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled=true --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool

# Source: retool/templates/deployment_workflows_worker.yaml
# Source: retool/templates/deployment_workflows_worker.yaml
```

Migration/upgrade scenarios:

**1) using a `retool-workflows-helm`, moving to `retool-helm` 6.0.2 ... pre Retool image 3.6.11**
- `workflows: enabled` would have been set; if it's not set for some reason, we won't spin up workers as expected.
- we continue to respect any other fields related to workflows as is (codeExecutor, retool-temporal-services-helm)

**2) using a `retool-workflows-helm`, moving to `retool-helm` 6.0.2 ... post Retool image 3.6.11** 
- `workflows: enabled` would have been set; if it's not set for some reason, we will spin up workers + backend (but worker will not CrashLoopBackOff)
- we continue to respect any other fields related to workflows as is (codeExecutor, retool-temporal-services-helm)

**3) using `retool-helm` <= 5.0.10, moving to `retool-helm` 6.0.2 ... pre Retool image 3.6.11**
- if you wanted workflows, you MUST set explicitly since these workers will CrashLoopBackOff without temporal config (connecting to external cluster) OR retool-temporal-services helm spinning up a local cluster.
- if you don't want workflows, you can set nothing -- and we will not spin up workflows.
- we continue to respect any other fields related to workflows as is (codeExecutor, retool-temporal-services-helm)

```
➜  retool git:(avimoondra/set-min-versions-for-workflow-startup) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.6.10" --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool
```

```
➜  retool git:(avimoondra/set-min-versions-for-workflow-startup) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.6.10" --set workflows.enabled=false  --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool
```

```
➜  retool git:(avimoondra/set-min-versions-for-workflow-startup) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.6.10" --set workflows.enabled=true --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool

# Source: retool/templates/deployment_workflows_worker.yaml
# Source: retool/templates/deployment_workflows_worker.yaml
```

**4) using `retool-helm` <= 5.0.10, moving to `retool-helm` 6.0.2 ... post Retool image 3.6.11**
- if you wanted workflows, you can either set nothing OR set explicitly, we will spin up backend +  worker, and worker will not CrashLoopBackOff
- if you don't want workflows, you must explicitly set `workflows: enabled` to `false`
- we continue to respect any other fields related to workflows as is (codeExecutor, retool-temporal-services-helm)

```
➜  retool git:(avimoondra/set-min-versions-for-workflow-startup) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.6.12" --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool

# Source: retool/templates/deployment_workflows_worker.yaml
# Source: retool/templates/deployment_workflows_worker.yaml
```

```
➜  retool git:(avimoondra/set-min-versions-for-workflow-startup) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.6.12" --set workflows.enabled=true --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool

# Source: retool/templates/deployment_workflows_worker.yaml
# Source: retool/templates/deployment_workflows_worker.yaml
```

```
➜  retool git:(avimoondra/set-min-versions-for-workflow-startup) ✗ helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.6.12" --set workflows.enabled=false --debug | grep _worker.yaml
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/avimoondra/retool-helm/charts/retool
```

### CODE REVIEW

File: charts/retool/templates/_helpers.tpl [hunk 1]
```diff
@@ -123,6 +123,30 @@ Set postgresql user
...
+  {{- $output = "" -}}
+{{- else if eq .Values.image.tag "latest" }}
+  {{- $output = "1" -}}
+{{- else if semverCompare ">= 3.6.11" .Values.image.tag }}
+  {{- $output = "1" -}}
+{{- else }}
+  {{- $output = "" -}}
+{{- end }}
+{{ $output }}
+{{- end }}
```
Comment (avimoondra) [inline charts/retool/templates/_helpers.tpl:148]: https://github.com/helm/helm/issues/10382

File: charts/retool/templates/_helpers.tpl [hunk 2]
```diff
@@ -123,6 +123,30 @@ Set postgresql user
...
+{{- else if empty .Values.image.tag -}}
+  {{- $output = "" -}}
+{{- else if eq .Values.image.tag "latest" -}}
+  {{- $output = "1" -}}
+{{- else if semverCompare ">= 3.6.11" .Values.image.tag -}}
+  {{- $output = "1" -}}
+{{- else -}}
+  {{- $output = "" -}}
+{{- end -}}
+{{- $output -}}
```
Comment (avimoondra) [inline charts/retool/templates/_helpers.tpl:147]: https://github.com/helm/helm/issues/10382

File: charts/retool/templates/_helpers.tpl
Comment (avimoondra) [inline charts/retool/templates/_helpers.tpl:1]: inspired by https://github.com/helm/helm/issues/10382

File: charts/retool/templates/deployment_workflows_worker.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.workflows.enabled }}
+{{- if include "retool.workflows.enabled" . }}
```
Comment (ryanartecona) [inline charts/retool/templates/deployment_workflows_worker.yaml:1]: whoa, this works? how does it coerce to boolean, output empty vs nonempty?

**Review by ryanartecona — COMMENTED** [review]
thanks for the description, that helped a ton understand what was happening. makes sense!

**Review by ryanartecona — APPROVED** [review]

File: charts/retool/templates/deployment_workflows_worker.yaml (hunk 3 above)
Comment (avimoondra) [reply charts/retool/templates/deployment_workflows_worker.yaml:1]: yup, exactly! https://helm.sh/docs/chart_template_guide/control_structures/#:~:text=A%20pipeline%20is%20evaluated%20as,an%20empty%20string

**Pull request merged**